runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts

```diff
@@ -193,6 +227,20 @@ export default class ExcelActionHandler extends StatefulActionHandler {
         }
     }
 
+
+    private isApplicationOpen() {
+        if (!this.session) {
+            throw new Error('There is no active Excel session. Open application before');
+        }
+    }
+
+    private checkWorksheetAndThrowError(worksheet: string, shouldExist: boolean): void {
+        if ((shouldExist && !this.checkIfWorksheetExist(worksheet)) ||
+            (!shouldExist && (worksheet.trim() === "" || this.checkIfWorksheetExist(worksheet)))) {
+                throw new Error(ExcelErrorMessage.existenceOrAbsenceOfWorksheet(shouldExist));
+        }
+    }
```

Comment:
name of the function suggests that regardless of the check outcome it will throw an error. Imo something like `checkNewWorksheet` will fit better, you can include info about possible error in the comment above